### ISSUE TITLE
source domain pre-training

### ISSUE THREAD

**Issue by hahapt** [zkcys001/UDAStrongBaseline#5]
In the source domain pre-training stage, I found it take target domain data to the model just for forward(trainers.py l137), I am confusing about why do this?
`# target samples: only forward`
 `_,_,_= self.model(t_inputs,training=True)`

**Comment by zkcys001** [comment]
Because this project is designed for the UDA task, thus we focus on the performance of the target domain.
Specifically, the forward operation is to calculate the BN parameter for obtaining the statistics of the target domain. It is beneficial for the target domain

**Comment by hahapt** [comment]
Sorry I'm still a little confusing. 
Here we only take the target domain data forward and return nothing, and here is no loss back propagation. So how do we calculate the BN parameter and it is beneficial for the feature clustering and fine-tune?

**Comment by zkcys001** [comment]
Q1: How do we calculate the BN parameter
R1: My response is not accurate. The update of the BN (mean and variance) does not require loss backpropagation. When you forward the data, BN layer can calculate the mean and variance, and update these. But the \beta and \gamma are not updated.

Q2: It is beneficial for the feature clustering and fine-tune
R2: It is a little beneficial for the initialization of feature clustering.

**Comment by zkcys001** [comment]
Loss back propagation for the udapte of  \beta and \gamma
Forward for the udapte of  mean and variance

**Comment by hahapt** [comment]
Thank you I got it!

**Comment by hahapt** [comment]
Thank you I got it!
